tests/test.d

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env rdmd
+/// Port of `test.sh`. Runs the tests in this directory.
+import std.algorithm, std.array, std.conv, std.file, std.path, std.process;
+import std.stdio, std.string, std.typecons, std.range, std.uni;
+
+version (Windows)
+    enum dfmt = "..\\bin\\dfmt.exe";
+else
+    enum dfmt = "../bin/dfmt";
+
+int main()
+{
+    foreach (braceStyle; ["allman", "otbs", "knr"])
+    {
+        foreach (entry; dirEntries(".", "*.d", SpanMode.shallow).filter!(e => e.baseName(".d") != thisExePath.baseName(".exe")))
+        {
+            const source = entry.baseName;
+            const argsFile = source.stripExtension ~ ".args";
+            string[] args = argsFile.exists ? readText(argsFile).splitter!isWhite.filter!(a => a.length).array : [];
+            writeln(dfmt, " --brace_style=", braceStyle, " ", args.map!(a => a ~ " ").join, source);
+            auto output = File(buildPath(braceStyle, source ~ ".out"), "w+");
+            int dfmt_result = spawnProcess([dfmt, "--brace_style=" ~ braceStyle] ~ args ~ [source],
```

Comment:
`format`? It's easier to do logging when you avoid building process arguments in place using `~s`

Reply:
Thanks, I have reworked the logging, amongst other edits.